### PR TITLE
updating docs for improved server 2022 platform support

### PR DESCRIPTION
**Pull request by DavidSchott**
* Added more details on updated server 2022 platform enhancements
* Added section covering traffic redirection on Windows
* Updated requirements and limitations. Also moved them higher up to make them easier to find.
* Added more hyperlinks and pointers to docs

The codebase in this repo itself also needs to be updated for the new exemption settings. I have labelled the examples in the docs with the old creation date (October 2019).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,71 @@
...
+## Traffic redirection on Windows
+On Linux, iptables rules are used to intercept and redirect traffic. On Windows, this is achieved using the HNS [L4WfpProxyPolicySetting](https://docs.microsoft.com/en-us/virtualization/api/hcn/hns_schema#l4wfpproxypolicysetting). The `L4WfpProxyPolicySetting` utilizes the [Windows Filtering Platform](https://docs.microsoft.com/en-us/windows/win32/fwp/about-windows-filtering-platform) (WFP) for network traffic filtering and redirection.
+
+For demonstration, here is an example JSON of a `L4WfpProxyPolicySetting` that will ensure all (in/out)bound `TCP` traffic gets redirected to (proxy) ports `9091`/`9191` respectively. unless:
+  * The traffic has already been inspected by the network proxy (process launched with user identity [SID S-15-18](https://docs.microsoft.com/en-us/windows/security/identity-protection/access-control/security-identifiers)). This is used to distinguish traffic that has already been redirected and processed by the network proxy. This is required because otherwise traffic originating from the proxy would be forwarded back to the proxy, thereby creating a redirection loop.
+    * See `"UserSID"` field below.
```
Comment (pravb) [inline README.md:16]: Should we talk about GroupSIDs approach here due to the known issue of actual user SID being generated at runtime for Windows containers?

Comment (daschott) [reply README.md:16]: Sure, I can add another bullet point for that. 

Also, which SID do we recommend?
Is it S-1-5-32-556? That is the Builtin\Network Configuration Operators group

Comment (pravb) [reply README.md:16]: That is correct.

Comment (daschott) [reply README.md:16]: I have added a clarification; please review if that looks OK.

**Review by pravb — APPROVED** [review]

**Pull request merged**